WNPRC_Purchasing/src/org/labkey/wnprc_purchasing/WNPRC_PurchasingController.java

```diff
@@ -204,7 +202,11 @@ public Object execute(RequestForm requestForm, BindException errors) throws Exce
 
         private void sendRequestChangeEmailNotification(String oldStatus, EmailTemplateForm emailTemplateForm, RequestForm requestForm, @Nullable List<LineItem> oldLineItems) throws MessagingException, IOException, ValidationException
         {
-            List<User> labEndUsers = SecurityManager.getUsersWithPermissions(getContainer(), Collections.singleton(InsertPermission.class));
+            List<User> usersWithInsertPerm = SecurityManager.getUsersWithPermissions(getContainer(), Collections.singleton(InsertPermission.class));
+
+            //get the lab end user who originated the request
+            List <User> labEndUsers = usersWithInsertPerm.stream().filter(u -> u.getUserId() == emailTemplateForm.getRequester().getUserId()).collect(Collectors.toList());
+            User endUser = labEndUsers.size() == 1 && labEndUsers.get(0) != null ? labEndUsers.get(0) : null;
```

Comment:
Why do you need to check if labEndUsers.get(0) is null here?

Reply:
removed